stdlib/Distributed/test/distributed_exec.jl

```diff
@@ -1592,10 +1592,39 @@ function reuseport_tests()
     @test all(in(results), procs())
 end
 
+# Test failure during worker setup
+old_stderr = stderr
+stderr_out, stderr_in = redirect_stderr()
+try
+    (nprocs() > 1) && rmprocs(workers())
+    npids = addprocs(1; topology=:all_to_all, lazy=false)
+    @test length(npids) == 1
+    @test nprocs() == 2
+    w2_connect_at = Distributed.PGRP.workers[2].config.connect_at
+
+    # test condition where connect fails immediately
+    lsock = listenany(ip"127.0.0.1", 20000)
+    Distributed.PGRP.workers[2].config.connect_at = ("127.0.0.1", lsock[1])
+    close(lsock[2])
+    npids = addprocs_with_testenv(1; topology=:all_to_all, lazy=false)
+    @test length(npids) == 0
+    @test nprocs() == 2
+
+    # test condition where connect times out
+    # using a non routable IP to test, ref: https://tools.ietf.org/html/rfc5737
+    Distributed.PGRP.workers[2].config.connect_at = ("203.0.113.0", w2_connect_at[2])
+    npids = addprocs_with_testenv(1; topology=:all_to_all, lazy=false)
```

Comment:
Default worker timeout if ENV `"JULIA_WORKER_TIMEOUT"` is unset is 60.0 seconds. This will unnecessarily delay the test suite run. Better to test with a smaller value - see an example in https://github.com/JuliaLang/julia/blob/b7774095f62e47bb2c8467b8a328075ab5a3fb89/stdlib/Distributed/test/distributed_exec.jl#L1149 and 

https://github.com/JuliaLang/julia/blob/b7774095f62e47bb2c8467b8a328075ab5a3fb89/stdlib/Distributed/test/distributed_exec.jl#L1159-L1161

which sets JULIA_WORKER_TIMEOUT to 1 second for one of the testruns.

Reply:
Done, `addprocs` times out quick now. But `rmprocs()` still takes a bit of time when the worker is stuck.

Reply:
Not sure if issuing a `kill` to the stuck test worker process before `rmprocs` a good idea? It does seem to work, but throws an ugly stacktrace out to stderr.

Reply:
And it was not difficult to avoid that. Added a commit with that now.